src/oom/oomd-util.c

```diff
@@ -0,0 +1,352 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <sys/xattr.h>
+#include <unistd.h>
+
+#include "fd-util.h"
+#include "format-util.h"
+#include "oomd-util.h"
+#include "parse-util.h"
+#include "path-util.h"
+#include "procfs-util.h"
+#include "signal-util.h"
+#include "stat-util.h"
+#include "stdio-util.h"
+
+static int log_kill(pid_t pid, int sig, void *userdata) {
+        log_debug("oomd attempting to kill " PID_FMT " with %s", pid, signal_to_string(sig));
+        return 0;
+}
+
+static int increment_oomd_xattr(const char *path, const char *xattr, int num_procs_killed) {
+        _cleanup_free_ char *value = NULL;
+        char buf[DECIMAL_STR_MAX(int) + 1];
+        int r, curr_count = 0;
+
+        assert(path);
+        assert(xattr);
+
+        r = cg_get_xattr_malloc(SYSTEMD_CGROUP_CONTROLLER, path, xattr, &value);
+        if (r < 0 && r != -ENODATA)
+                return r;
+
+        if (!isempty(value)) {
+                 r = safe_atoi(value, &curr_count);
+                 if (r < 0)
+                         return r;
+        }
+
+        xsprintf(buf, "%d", curr_count + num_procs_killed);
```

Comment:
hmm, should we maybe do some overflow check here? i.e. return EOVERFLOW or so if this goes beyond INT_MAX?